### PR TITLE
Refactoring read orientation model to run within Mutect2

### PR DESCRIPTION
**Pull request by davidbenjamin**
@takutosato Most of the files changed are just due to a change in method signature.  Any significant block of "new" code is just your code moved to a different class.  This PR does a few things:

* Move logic from the orientation bias annotation into the filter.
* Package F1R2 counts and learned orientation bias models in .tar.gz files to simplify command lines and accomodate multiple samples.
* Make all orientation bias tools fully multi-sample.
*Extract a backend for CollectF1R2Counts and use this backend inside Mutect2.

I have tested the new pipeline on Firecloud.

Do you have time to review this before the release tomorrow?  If not, I can ask Lee.

### CODE REVIEW

**Comment by takutosato** [comment]
@davidbenjamin I will review this today. Thanks for doing this refactoring.

**Comment by davidbenjamin** [comment]
@takutosato Thanks for having the idea!  It was really fun seeing the Firecloud timing diagram go straight from SplitIntervals to M2!  Maybe a little too fun.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/readorientation/F1R2CountsCollector.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    // alt table writer for each sample
+    private Map<String, AltSiteRecord.AltSiteRecordTableWriter> altTableWriters;
+
+    private final File outputTarGzFile;
+
+    private final File tmpDir = IOUtils.createTempDir("untarred");
+
+    public F1R2CountsCollector(final CollectF1R2CountsArgumentCollection CF1R2Args, final SAMFileHeader header, final File outputTarGzFile) {
+        this.CF1R2Args = CF1R2Args;
+        samples = ReadUtils.getSamplesFromHeader(header);
```
Comment (takutosato) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/readorientation/F1R2CountsCollector.java]: We should limit the samples to tumor samples here. The collector stores normal samples in `samples`, but the actual data collection is done only on the tumor samples, which down the line creates an empty prior table for the normal samples in `LearnReadOrientationModel.`

File: scripts/mutect2_wdl/mutect2.wdl [hunk 2]
```diff
@@ -40,9 +36,9 @@
 ## normal_bam, normal_bam_index: BAM and index for the normal sample
 ##
 ## ** Primary resources ** (optional but strongly recommended)
-## pon, pon_index: optional panel of normals in VCF format containing probable technical artifacts (false positves)
```
Comment (takutosato) [inline scripts/mutect2_wdl/mutect2.wdl]: I think these index files are needed for the regular Mutect2.wdl, right?

Comment (takutosato) [reply scripts/mutect2_wdl/mutect2.wdl]: Also we should use either the `_idx` suffix or `_index`, not both

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/readorientation/F1R2CountsCollector.java (hunk 1 above)
Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/readorientation/F1R2CountsCollector.java]: done

File: scripts/mutect2_wdl/mutect2.wdl (hunk 2 above)
Comment (davidbenjamin) [reply scripts/mutect2_wdl/mutect2.wdl]: done x 2

**Comment by takutosato** [comment]
Evaluation finished. It looks good!

**Pull request merged**